macrosynergy/panel/hedge_ratio.py

```diff
@@ -1,39 +1,274 @@
+
 import numpy as np
 import pandas as pd
 from typing import List, Union
+import statsmodels.api as sm
+from macrosynergy.management.simulate_quantamental_data import make_qdf
+from macrosynergy.management.shape_dfs import reduce_df
+
+
+def refreq_groupby(start_date: pd.Timestamp, end_date: pd.Timestamp, refreq: str = 'm'):
+    """
+    The hedging ratio is re-estimated according to the frequency parameter. Therefore,
+    break up the respective return series, which are defined daily, into the re-estimated
+    frequency paradigm. To achieve this ensure the dates produced fall on business days,
+    and will subsequently be present in the return-series dataframes.
+
+    :param <pd.Timestamp> start_date:
+    :param <pd.Timestamp> end_date:
+    :param <str> refreq:
+
+    return <List[pd.Timestamp]>: List of timestamps where each date is a valid business
+        day, and the gap between each date is delimited by the frequency parameter.
+    """
 
-def hedge_ratio(df: pd.DataFrame, xcats: List[str] = None, cids: List[str] = None,
-                hedge_return: str = None,
-                start: str = None, end: str = None,
-                meth: str = 'ols', oos: bool = True, rfreq: str = 'm',
-                minobs: int = 24,
-                blacklist: dict = None):
+    dates = pd.date_range(start_date, end_date, freq=refreq)
```

Comment:
The ratio needs to be calculated based on daily data (for now). The rebalancing frequency only affects the intervals of recalculation.